### PR TITLE
[WIP] Add debug commands for repoobj id_hash/parse/format 

### PR DESCRIPTION
**Pull request by shalearkane**
Implements #7406
- [x] id-hash
- [ ] parse-obj
- [ ] format-obj
- [ ] Tests

### CODE REVIEW

File: src/borg/archiver/debug_cmd.py [hunk 1]
```diff
@@ -294,6 +294,23 @@ def do_debug_delete_obj(self, args, repository):
...
+        with open(args.path, "rb") as f:
+            data = f.read()
+
+        from ..crypto.key import key_factory
+
+        # set up the key without depending on a manifest obj
+        ids = repository.list(limit=1, marker=None)
+        cdata = repository.get(ids[0])
+        key = key_factory(repository, cdata)
+        repo_objs = RepoObj(key)
```
Comment (ThomasWaldmann) [inline src/borg/archiver/debug_cmd.py:309]: if you used `manifest=True`, would you get this "for free"?

File: src/borg/archiver/debug_cmd.py [hunk 2]
```diff
@@ -545,6 +562,24 @@ def build_parser_debug(self, subparsers, common_parser, mid_common_parser):
             "ids", metavar="IDs", nargs="+", type=str, help="hex object ID(s) to delete from the repo"
         )
 
+        debug_id_hash_obj_epilog = process_epilog(
+            """
+            This command reads a plaintext file and computes object id from it
```
Comment (ThomasWaldmann) [inline src/borg/archiver/debug_cmd.py:567]: This command computes the id-hash for some file content.

(from 1.2-maint)

File: src/borg/archiver/debug_cmd.py [hunk 3]
```diff
@@ -294,6 +294,23 @@ def do_debug_delete_obj(self, args, repository):
         print("Done.")
         return EXIT_SUCCESS
 
+    @with_repository(manifest=False, exclusive=True)
+    def do_debug_id_hash_file(self, args, repository):
+        """calculate object id of a given plaintext file"""
```
Comment (ThomasWaldmann) [inline src/borg/archiver/debug_cmd.py:299]: """compute id-hash for file contents"""

(from 1.2-maint)

File: src/borg/archiver/debug_cmd.py [hunk 4]
```diff
@@ -545,6 +562,24 @@ def build_parser_debug(self, subparsers, common_parser, mid_common_parser):
...
+            parents=[common_parser],
+            add_help=False,
+            description=self.do_debug_id_hash_file.__doc__,
+            epilog=debug_id_hash_obj_epilog,
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            help="calculate object id from plaintext file (debug)",
+        )
+
+        subparser.set_defaults(func=self.do_debug_id_hash_file)
+        subparser.add_argument("path", metavar="PATH", type=str, help="plaintext file to calculate object id from")
```
Comment (ThomasWaldmann) [inline src/borg/archiver/debug_cmd.py:581]: help='content for which the id-hash shall get computed')

(from 1.2-maint)

**Review by ThomasWaldmann — CHANGES_REQUESTED** [review]
Looks good overall, but I guess we can simplify it a bit and also make the wording consistent with borg 1.2.x.

**Comment by ThomasWaldmann** [comment]
hi @shalearkane, do you still work on this?

guess you can ignore my feedback about id-hash and have a look at #7436 - looks like there was some parallel work and they already solved the id-hash part.

i guess i can merge #7436 soon and maybe easiest for you is to update your local master branch and then start a new PR, if you want to work on parse/format.

**Comment by shalearkane** [comment]
Yes definitely. I am closing this pull request and opening another.

**Pull request closed without merging**